src/InstrumentationEngine.Attach/Program.cs

```diff
@@ -136,26 +225,91 @@ internal static int Main(string[] args)
                 return ExitCodeFailure;
             }
 
+            #endregion
+
+            #region Profiler Attach
             // CONSIDER: Should the engine be validated before attempting to attach it to the CLR?
             // For example, could check the signature on the assembly (which wouldn't work on Linux).
 
             DiagnosticsClient client = new DiagnosticsClient(processId);
             try
             {
-                client.AttachProfiler(TimeSpan.FromSeconds(10), InstrumentationEngineClsid, enginePath);
+                client.AttachProfiler(TimeSpan.FromSeconds(10), InstrumentationEngineClsid, enginePath, bytes);
             }
             catch (ServerErrorException ex)
             {
                 WriteError(Invariant($"Could not attach engine to process: '{ex.Message}'."));
                 return ExitCodeFailure;
             }
 
+            #endregion
+
             return ExitCodeSuccess;
         }
 
         private static void WriteError(string message)
         {
             Console.Error.WriteLine(message);
         }
+
+        private static bool TryParseEngineConfiguration(
+            InstrumentationConfigurationSources sources,
+            Chip targetChip,
+            out InstrumentationEngineConfiguration configuration)
+        {
+            var engineSection = new InstrumentationEngineConfigurationInstrumentationEngine();
+            var methodsSection = new List<instrumentationMethodsTypeAddInstrumentationMethod>();
+
+            // Parse engine settings
+            var engineSettings = new List<settingsTypeSetting>()
+            {
+                new settingsTypeSetting()
+                {
+                    Name = "LogLevel",
+                    Value = "Errors"
+                }
+            };
+            engineSection.Settings = engineSettings.ToArray();
+
+            // Parse methods section
+            foreach (var source in sources.InstrumentationConfigurationSource)
+            {
+                foreach (var platform in source.Platforms)
+                {
+                    if (platform.Chip.Equals(targetChip.ToString("G"), StringComparison.OrdinalIgnoreCase))
+                    {
+                        var methodSettings = new List<settingsTypeSetting>();
+                        var methodSection = new instrumentationMethodsTypeAddInstrumentationMethod()
+                        {
+                            ConfigPath = platform.Path
```

Comment:
>platform.Path [](start = 41, length = 13)

This path is relative to the folder that the xml file was found in. So you should use Path.Combine to prepend the that folder to this path and then call Path.GetFullPath to resolve it.

Reply:
Nice, Path.Combine will ignore previous arguments if an argument is a fullpath.

---
In reply to: [388452994](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r388452994) [](ancestors = 388452994)

Reply:
Can also add a Path.IsPathRooted check before doing Path.Combine, but in practice, I don't think this value is ever going to be an absolute path.

---
In reply to: [388485290](https://github.com/microsoft/clrinstrumentationengine/pull/220#discussion_r388485290) [](ancestors = 388485290,388452994)

Reply:
Path.GetFullPath just ensures the path is absolute but doesn't check for existence, I'll add a File.Exists check

---
In reply to: [388485290](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r388485290) [](ancestors = 388485290,388452994)